crates/trigger-http/src/lib.rs

```diff
@@ -475,11 +492,21 @@ fn parse_listen_addr(addr: &str) -> anyhow::Result<SocketAddr> {
 
 /// The incoming request's scheme and authority
 ///
-/// The incoming request's URI is relative to the server, so we need to set the scheme and authority
-fn set_req_uri(req: &mut Request<Body>, scheme: Scheme, addr: SocketAddr) -> Result<()> {
+/// The incoming request's URI is relative to the server, so we need to set the scheme and authority.
+/// The `Host` header is used to set the authority. This function will error if no `Host` header is
```

Comment:
> The `Host` header is used to set the authority.

Should this also check the `req.uri().authority()`? I think it _might_ be most correct to use either one, verifying that they are identical if both are present.

Reply:
I'm pretty sure that's right, based on https://github.com/hyperium/hyper/issues/1612